### ISSUE TITLE
function returns address of local variable [-Wreturn-local-addr]

### ISSUE THREAD

**Issue by sheerluck** [polymake/polymake#6]
$ gcc --version
gcc (Gentoo 9.1.0 p1.0) 9.1.0

in «pm::Rational&& pm::operator>>(pm::Rational&&, long int)»:
polymake-3.4/include/core/polymake/Rational.h:1356:26
function returns address of local variable [-Wreturn-local-addr]
```
 1356 |       return std::move(a >> k);
      |                        ~~^~~~
```

**Comment by benlorenz** [comment]
Thanks for the report, this line should be `return std::move(a >>= k);` instead. There are a few more minor things to make polymake compile with gcc 9.1 but we are working on that.

**Comment by benlorenz** [comment]
This is fixed in polymake 3.5.
